core/src/main/java/org/jdbi/v3/core/statement/internal/DefineBindsTemplateEngine.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jdbi.v3.core.statement.internal;
+
+import java.util.HashSet;
+import java.util.Set;
+
+import org.jdbi.v3.core.argument.NullArgument;
+import org.jdbi.v3.core.statement.StatementContext;
+import org.jdbi.v3.core.statement.TemplateEngine;
+
+public class DefineBindsTemplateEngine implements TemplateEngine {
+    private final TemplateEngine delegate;
+
+    public DefineBindsTemplateEngine(TemplateEngine delegate) {
+        this.delegate = delegate;
+    }
+
+    @Override
+    public String render(String template, StatementContext ctx) {
+        final Set<String> binds = new HashSet<>(ctx.getBinding().getNames());
+        binds.removeAll(ctx.getAttributes().keySet());
+        for (String boundButNotDefined : binds) {
+            ctx.getBinding().findForName(boundButNotDefined, ctx)
+                .map(a -> a instanceof NullArgument ? null : a)
+                .ifPresent(a -> ctx.define(boundButNotDefined, a));
```

Comment:
```java
String name = "Robert'); drop table students;--";
handle.createUpdate(sql)
    .bind("name", name)
    .execute();
```

Making an unsanitized argument value available to the template engine is a major SQL injection attack vector.

What do you think about defining attribute value instead as a boolean `false` if the argument is null, `true` if non-null:

```java
ctx.getBinding().findForName(boundButNotDefined, ctx)
   .ifPresent(a -> ctx.define(boundButNotDefined, !(a instanceof NullArgument)));
```

This way a full-fledged template engine has a Boolean property to key off of, but we're not opening up our users to get pwned by a maliciously crafted parameter.

Reply:
It's true, all I really need is the presence or absence of the value.  I could make it an option, with the default to just bind a boolean.  Or I could just hardcode it to always map to boolean, which is less useful generally, but still solves my particular use case.

Reply:
I've made it configurable, and the default is to only bind it as a boolean.  Hopefully that's in line with your suggestion.

Reply:
I'm ok with that

Reply:
Please add a warning about SQL injection to the interface and the config parameter setter so implementers are given every opportunity not to shoot themselves in the foot.